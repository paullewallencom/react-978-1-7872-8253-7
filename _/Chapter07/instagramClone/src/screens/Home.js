import React from 'react';
import { getMessages, postMessage } from '../services/api';
import { Image, View, KeyboardAvoidingView, Platform, Text, StyleSheet, Button, FlatList } from 'react-native';
import { authDecorator } from '../services/authService';

class Home extends React.Component {

  keyboardVerticalOffset = Platform.OS === 'ios' ? 60 : 0


  render(){
    return (
      <View style={styles.container}><Text>Home</Text></View>
    )
  }
}

export default authDecorator(Home)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
  }
})
