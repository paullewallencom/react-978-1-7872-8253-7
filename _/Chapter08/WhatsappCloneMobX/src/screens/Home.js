import React from 'react';
import { View, Button, StyleSheet } from 'react-native';

export default class Home extends React.Component {

  static navigationOptions = {
    title: "Home Screen"
  }

  render(){
    return (
      <View style={styles.container}>
        <Button title="Navigate to ChatScreen"
          onPress={() => this.props.navigation.navigate('chat', { name: 'John' })}/>
      </View>
    )
  }
}

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: 'white',
      justifyContent: 'center',
      alignItems: 'center'
    }
  })
