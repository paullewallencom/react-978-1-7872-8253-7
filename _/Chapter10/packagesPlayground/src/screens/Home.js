import React from 'react';
import { View, FlatList, StyleSheet } from 'react-native';
import HomeScreenNavItem from '../components/HomeScreenNavItem';

const MENU = [
    { 
        caption: 'Shoutem',
        routeName: 'shoutem',
        iconName: "mobile"
    },
    {
        caption: 'NativeBase',
        routeName: 'nativeBase',
        iconName: 'mobile'
    },
    {
        caption: 'react-native-maps',
        routeName: 'maps',
        iconName: 'map-marker'
    },
    {
        caption: 'Image Picker',
        routeName: 'imagepicker',
        iconName: 'image'
    },
    {
        caption: 'Video',
        routeName: 'video',
        iconName: 'tv'
    }, 
    {
        caption: 'Toasts',
        routeName: 'toasts',
        iconName: 'comment'
    },
    {
        caption: 'Camera',
        routeName: 'camera',
        iconName: 'camera'
    },
    {
        caption: 'Push Notifications',
        routeName: 'pn',
        iconName: 'notification',
        isEntypo: true
    }
]

export default Home = ({ navigation }) => (
    <FlatList
        style={styles.container}
        data={MENU}
        renderItem={({ item }) => (
            <HomeScreenNavItem { ...item } navigate={navigation.navigate} />
        )}
        keyExtractor={(item, index) => (`navItem-${index}`)}
    />
)
Home.navigationOptions = ({ navigation }) => ({
    title: "Selected packages list"
})

const styles = StyleSheet.create({
    container: {
        flex: 1
    }
})