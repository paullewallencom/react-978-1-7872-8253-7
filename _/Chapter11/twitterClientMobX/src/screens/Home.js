import React from "react";
import {
  View,
  StatusBar
} from "react-native";
import { List, ListItem, Card, Icon } from 'react-native-elements';
import { Feed } from "../components/Feed";
import { ProfileHeaderButton } from '../components/ProfileHeaderButton';


//https://dev.twitter.com/rest/reference/get/statuses/home_timeline
 

export class Home extends React.Component {
  static navigationOptions = ({ navigation}) => ({
    title: "Home",
    headerBackTitle: null,
    headerLeft: <ProfileHeaderButton navigation={navigation} />,
  });

  
  render() {
    const { navigation } = this.props;  
    return (
      <View style={{ flex: 1, backgroundColor: "white" }}>
        <StatusBar translucent barStyle="dark-content" />
        <Feed endpoint="statuses/home_timeline" navigation={navigation} />
      </View>
    )
  }
}
