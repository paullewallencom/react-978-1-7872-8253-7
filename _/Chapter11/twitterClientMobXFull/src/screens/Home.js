import React from "react";
import {
  View,
  StatusBar
} from "react-native";
import { List, ListItem, Card, Icon } from 'react-native-elements';
import { Feed } from "../components/Feed";
import { ProfileHeaderButton } from '../components/ProfileHeaderButton';
import { observer, inject } from 'mobx-react';
 

@inject('userStore')
@observer
export class Home extends React.Component {
  static navigationOptions = ({ navigation}) => ({
    title: "Home",
    headerBackTitle: null,
    headerLeft: <ProfileHeaderButton navigation={navigation} />,
  });



  
  render() {
    const { navigation, userStore } = this.props;  
    return (
      userStore.isLoggedIn && 
      <View style={{ flex: 1, backgroundColor: "white" }}>
        <StatusBar translucent barStyle="dark-content" />
        <Feed endpoint="statuses/home_timeline" navigation={navigation} />
      </View>
    )
  }
}
