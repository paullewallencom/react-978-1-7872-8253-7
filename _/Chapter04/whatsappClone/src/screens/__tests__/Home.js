import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import Home from '../Home';

it('renders correctly', () => {
    const tree = renderer.create(
        <Home />
    );
    expect(tree).toMatchSnapshot();
});

it('has proper navigation options', () => {
    expect(Home.navigationOptions).toMatchSnapshot();
})

it('navigate on press', () => {
    const navigate = jest.fn();
    const homeScreen = shallow(
        <Home navigation={{ navigate }} />
    )
    homeScreen.find('Button').simulate('press');
    expect(navigate).toBeCalledWith("chat", { "name": "John" });
})