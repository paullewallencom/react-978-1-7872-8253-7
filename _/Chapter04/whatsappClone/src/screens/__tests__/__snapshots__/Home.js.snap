// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has proper navigation options 1`] = `
Object {
  "title": "Home Screen",
}
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Navigate to ChatScreen
        </Text>
      </View>
    </View>
  </View>
</View>
`;
